File: Data Structures & Algorithms/three-integer-sum/submission-15.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        Arrays.sort(nums);

        for(int i = 0; i < nums.length; i++){
            int current = nums[i];
            if(i > 0 && current == nums[i -1]) continue;

            int left = i + 1;
            int right = nums.length -1;


            while(left < right){
                int currentSum = current + nums[left] + nums[right];

                if(currentSum > 0){
                    right--;
                }else if(currentSum < 0){
                    left++;
                }else {
                    result.add(List.of(current, nums[left], nums[right]));
                    left++;
                    right--;

                    while(left < right && nums[left] == nums[left -1]){
                        left++;
                    }
                }
            }
        } 

        return result;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-15.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        most_repeated = 0
        visited = defaultdict()
        result = 0

        left = 0
        right = 0

        while right < len(s):
            current = s[right]

            visited[current] = visited.get(current, 0) + 1

            most_repeated = max(most_repeated, visited[current])

            while ((right - left) + 1) - most_repeated > k:
                current_left = s[left]
                visited[current_left] = visited[current_left] - 1
                left += 1
            
            result = max((right - left) + 1, result)
            right += 1
        
        return result

            

            

